app/scripts/metamask-controller.js

```diff
@@ -1453,6 +1453,9 @@ export default class MetamaskController extends EventEmitter {
       removeSnapError: this.snapController.removeSnapError.bind(
         this.snapController,
       ),
+      disableSnap: this.snapController.disableSnap.bind(this.snapController),
+      enableSnap: this.snapController.enableSnap.bind(this.snapController),
+      removeSnap: this.snapController.removeSnap.bind(this.snapController),
```

Comment:
```suggestion
      removeSnap: this.snapController.removeSnap.bind(this.snapController),
      revokePermission: this.permissionController.revokePermission.bind(this.permissionController),
```

Reply:
we need to expose reveokPermission so the ui can call it

Reply:
or `snapController.removeAllPermissionsFor` (they both do the same thing, removeAllPermissionFor calls revoke underneath)